File: choresBackend/src/main/java/com/ori/chores/Beans/Chore.java
package com.ori.chores.Beans;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;
import org.springframework.boot.context.properties.bind.DefaultValue;

import java.sql.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Chore {
//    public Chore() {
//        isDone = false;
//    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(nullable = false)
    private String choreName;
//    @Column(nullable = false)
    @OneToOne
    private Responsible responsible;
    private Date endDate;
    private Date doneDate;
    @Builder.Default
    private Boolean isDone=false;


}

File: choresAppFrontEnd/src/Components/Models/Chore.ts
import { Responsible } from "./Responsible";

export class Chore{
    public id:number;
    public choreName:string;
    public endDate:Date;
    public doneDate:Date|null;
    public isDone:Boolean;
    public responsible:Responsible;
    
    constructor( id:number,choreName:string,isDone:Boolean,responsible:Responsible,endDate:Date,doneDate:Date){
            this.id=id;
            this.choreName=choreName;
            if(doneDate!=null){
                this.doneDate=doneDate;
            }else{this.doneDate=null}
            this.endDate=endDate;
            this.isDone=isDone;
            this.responsible=responsible;
        }
       
}
